### ISSUE TITLE
JavaScript gets corrupted (newline removed after trailing tabs)

### ISSUE THREAD

**Issue by spasche** [zaproxy/zaproxy#4630]
Hello,

I'm using Zap on an application that outputs JavaScript inside the index page. The proxy changes the original JavaScript in a way that breaks it. I tested with Zap for Windows version 2.7.0 and also the latest weekly release ZAP_WEEKLY_D-2018-04-23.

Here's the request done by the browser:
```
GET /xxx/yyy HTTP/1.1
Host: xxx:ppp
Connection: keep-alive
Cache-Control: max-age=0
Upgrade-Insecure-Requests: 1
User-Agent: Mozilla/5.0 (Windows NT 6.3; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/64.0.3282.167 Safari/537.36
Accept: text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Cookie: jsessionid_xxx=yyy

```

Original response from server, without ZAP
```
HTTP/1.1 200 OK
Content-Language: en-US
Content-Type: text/html;charset=UTF-8
Transfer-Encoding: chunked
Connection: keep-alive

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
[...]
		<script type="text/javascript">/* <![CDATA[ */
[...]
				// some comment FOLLOWED BY TRAILING TABS:		
				theLineAfterwards();
```

Response when going through ZAP:
```
HTTP/1.1 200 OK
Content-Language: en-US
Content-Type: text/html;charset=UTF-8
Connection: close

<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
[...]
<script type="text/javascript">/* <![CDATA[ */
[...]
// some comment FOLLOWED BY TRAILING TABS:theLineAfterwards();

```

The response contains TAB characters for indentation.
Basically, Zap removed the trailing tabs and newline on the comment line.
Consequently, the code line following the comment is not executed and the application doesn't work.

I tried to save the index HTML file locally in order to reproduce using a simple HTTP server, but I couldn't reproduce the issue. So at this stage I don't have a reduced test case.

**Comment by thc202** [comment]
Are there any replacer rules [1] or scripts [2] (or other functionality) enabled that could be changing the body? 

Does that also happen when sending the request directly from ZAP? [3]

[1] https://github.com/zaproxy/zap-extensions/wiki/HelpAddonsReplacerReplacer
[2] https://github.com/zaproxy/zap-core-help/wiki/HelpAddonsScriptsScripts
[3] https://github.com/zaproxy/zap-core-help/wiki/HelpUiDialogsMan_req

**Comment by spasche** [comment]
I forgot to mention that I ran Zap with its default configuration (proxy in standard mode).
So I don't have any replacer rules or scripts loaded that are changing the body.
The issue also happens if I run the request from the manual request editor dialog.

**Comment by thc202** [comment]
Could you provide a sample HTML file that reproduces the issue? (Might be related to Transfer-Encoding?)

**Comment by thc202** [comment]
Thanks, will check that then.

**Comment by thc202** [comment]
I was not able to reproduce this issue with 2.7.0 (nor with the latest network changes), I tried with no transfer encoding and with chunked, in either case the response was as expected (also tried with different newline variations `\r` `\n` `\r\n`, in case that was the problem but no success either).

Closing, if anyone is able to reproduce this, please, leave a comment with the steps to reproduce it.
